File: app/src/main/java/com/alvaro/proyectofinal/Juegos/Hangman/LetrasAdapter.java
package com.alvaro.proyectofinal.Juegos.Hangman;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.BaseAdapter;
import android.widget.Button;

import com.alvaro.proyectofinal.R;

public class LetrasAdapter extends BaseAdapter {

    private String[] letras;
    private LayoutInflater letrasLayout;

    public LetrasAdapter(Context context){
       letras = new String[26];
       for (int i = 0; i < letras.length; i++){
           letras[i] = "" + (char)(i + 'A');
       }
       letrasLayout = LayoutInflater.from(context);

    }

    @Override
    public int getCount() {
        return letras.length;
    }

    @Override
    public Object getItem(int i) {
        return null;
    }

    @Override
    public long getItemId(int i) {
        return 0;
    }

    @Override
    public View getView(int i, View view, ViewGroup viewGroup) {
        Button btnLetra;
        if (view == null){
            btnLetra = (Button) letrasLayout.inflate(R.layout.letras,viewGroup,false);
        }else {
            btnLetra = (Button)view;
        }
        btnLetra.setText(letras[i]);
        return btnLetra;
    }
}

File: app/src/main/java/com/alvaro/proyectofinal/Juegos/Hangman/LetrasAdapter.kt
package com.alvaro.proyectofinal.Juegos.Hangman

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.BaseAdapter
import android.widget.Button
import com.alvaro.proyectofinal.R

class LetrasAdapter(context: Context?) : BaseAdapter() {
    private val letras: Array<String?> = arrayOfNulls(26)
    private val letrasLayout: LayoutInflater

    init {
        for (i in letras.indices) {
            letras[i] = "" + (i + 'A'.code).toChar()
        }
        letrasLayout = LayoutInflater.from(context)
    }

    override fun getCount(): Int {
        TODO("Not yet implemented")
    }

    override fun getItem(i: Int): Any {
        TODO("Not yet implemented")
    }

    override fun getItemId(i: Int): Long {
        TODO("Not yet implemented")
    }

    override fun getView(i: Int, view: View, viewGroup: ViewGroup): View {
        val btnLetra: Button = view as Button
        btnLetra.text = letras[i]
        return btnLetra
    }
}